ui/app/serializers/volume.js

```diff
@@ -23,17 +23,16 @@ export default class VolumeSerializer extends ApplicationSerializer {
     hash.ID = JSON.stringify([`csi/${hash.ID}`, hash.NamespaceID || 'default']);
     hash.PluginID = `csi/${hash.PluginID}`;
 
-    // Convert hash-based allocation embeds to lists
+    // Populate read/write allocation lists from aggregate allocation list
     const readAllocs = hash.ReadAllocs || {};
     const writeAllocs = hash.WriteAllocs || {};
-    const bindIDToAlloc = hash => id => {
-      const alloc = hash[id];
-      alloc.ID = id;
+    const bindIDToAlloc = allocList => id => {
+      const alloc = allocList.find(a => a.ID === id);
       return alloc;
     };
 
-    hash.ReadAllocations = Object.keys(readAllocs).map(bindIDToAlloc(readAllocs));
-    hash.WriteAllocations = Object.keys(writeAllocs).map(bindIDToAlloc(writeAllocs));
+    hash.ReadAllocations = Object.keys(readAllocs).map(bindIDToAlloc(hash.Allocations));
+    hash.WriteAllocations = Object.keys(writeAllocs).map(bindIDToAlloc(hash.Allocations));
```

Comment:
Question for reviewers: would it save browser memory if we nulled-out the `hash.Allocations`, `hash.ReadAllocs` and `hash.WriteAllocs` fields after we're done with them here?

Reply:
I don't know off the top of my head if hash sticks around or if a copy is what gets inserted into the store. It wouldn't hurt to add a `delete hash.Allocations` just to err on the side of caution.